massdriver/resource_alarm_channel.go

```diff
@@ -0,0 +1,84 @@
+package massdriver
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type PackageAlarmMetadata struct {
+	ResourceIdentifier string `json:"resource_identifier"`
+	DisplayName        string `json:"display_name"`
+}
+
+func resourcePackageAlarm() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePackageAlarmCreate,
+		ReadContext:   schema.NoopContext,
+		UpdateContext: resourcePackageAlarmUpdate,
+		DeleteContext: resourcePackageAlarmDelete,
+
+		Schema: map[string]*schema.Schema{
+			"resource_identifier": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The identifier of the alarm. In Azure it will be the id, GCP will be the name, and in AWS it will be teh name",
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name to display in the massdriver UI",
+			},
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourcePackageAlarmCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*MassdriverClient)
+
+	var diags diag.Diagnostics
+
+	packageAlarmMeta := PackageAlarmMetadata{
+		ResourceIdentifier: d.Get("resource_identifier").(string),
+		DisplayName:        d.Get("display_name").(string),
+	}
+
+	event := NewEvent(EVENT_TYPE_ALARM_CHANNEL_CREATED)
+	event.Payload = EventPayloadAlarmChannels{DeploymentId: c.DeploymentID, PackageAlarm: packageAlarmMeta}
+
+	err := c.PublishEventToSNS(event, &diags)
+
+	if err != nil {
+		return diags
+	}
+
+	d.SetId(time.Now().Format(time.RFC3339))
+	d.Set("last_updated", time.Now().Format(time.RFC850))
+
+	return diags
+}
+
+func resourcePackageAlarmUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	d.SetId(time.Now().Format(time.RFC3339))
+	d.Set("last_updated", time.Now().Format(time.RFC850))
+
+	return diags
+}
+
+func resourcePackageAlarmDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	d.SetId(time.Now().Format(time.RFC3339))
```

Comment:
```suggestion
	d.SetId("")
```

See the above comment on the purpose of this field. In this case, since its a delete, we need to actually zero out the id field since that tells terraform that the resource no longer exists (and will remove it from state).